### PR TITLE
Speed up inspec

### PR DESCRIPTION
**Pull request by zenspider**
This attempts to shave off time from our tests by speeding up the implementation, mostly be rearranging when and how we load things. There's more to be done (I really want to remove the resource registration system!) but this should help us in the meantime.

### CODE REVIEW

File: lib/inspec/cli.rb [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
-require 'inspec/runner_mock'
-require 'inspec/env_printer'
-require 'inspec/schema'
-require 'inspec/config'
+require 'inspec/utils/deprecation/deprecator'
 require 'inspec/dist'
+require 'inspec/backend'
+require 'inspec/dependencies/cache'
+
+module Inspec # TODO: move this somewhere "better"?
```
Comment (miah) [inline lib/inspec/cli.rb:8]: I fear overloading the word 'core'; but maybe something like `lib/inspec/core.rb`?

File: test/functional/inspec_exec_test.rb [hunk 2]
```diff
@@ -8,6 +8,31 @@
     skip_windows!
   }
 
+#   it 'runs_directly' do
+#     require "inspec/cli"
+#     require "misc/exit_wtf"
```
Comment (miah) [inline test/functional/inspec_exec_test.rb]: Is that a debugging crumb?

**Review by miah — APPROVED** [review]

File: test/functional/inspec_exec_test.rb (hunk 2 above)
Comment (zenspider) [reply test/functional/inspec_exec_test.rb]: ah yes. I'll remove that.

File: lib/inspec/cli.rb (hunk 1 above)
Comment (zenspider) [reply lib/inspec/cli.rb:8]: I do too... doesn't have to happen in this commit. that's why I leave TODO's behind as I travel through the code.

**Comment by zenspider** [comment]
This is part of my optimization work, being tracked by #4212.

**Pull request merged**